src/sdk/pynni/nni/evolution_tuner/evolution_tuner.py

```diff
@@ -115,20 +101,69 @@ def update_search_space(self, search_space):
 
         self.random_state = np.random.RandomState()
         self.population = []
-        is_rand = dict()
-
-        for item in self.space:
-            is_rand[item] = True
-
-        for _ in range(self.population_size):
-            config = json2parameter(
-                self.searchspace_json, is_rand, self.random_state)
-            self.population.append(Individual(config=config))
 
+    def trial_end(self, parameter_id, success, **kwargs):
+        """
+        To deal with trial failure. If a trial fails, readd the trial parameters into population.
+        Parameters
+        ----------
+        parameter_id : int
+            Unique identifier for hyper-parameters used by this trial.
+        success : bool
+            True if the trial successfully completed; False if failed or terminated.
+        **kwargs
+            Not used
+        """
+        self.num_running_trials -= 1
+        logger.info('trial (%d) end')
+
+        if not success:
+            indv = self.running_trials[parameter_id]
+            self.population.append(indv)
```

Comment:
this config is failed, why add it back to `self.population`?

Reply:
Based on the discussion of the meeting, a failed trial may due to`out of memory`, so add the config back to the population.

Reply:
if the reason is out of memory, you run it again it will also fail due to out of memory

Reply:
If not deal with failed trials, the population would become smaller. Do you think generating a new random individual would be better?

Reply:
yes, randomly generating one is a possible option.